internal/v1/handler_blueprints.go

```diff
@@ -405,13 +405,28 @@ func (h *Handlers) GetBlueprint(ctx echo.Context, id openapi_types.UUID, params
 	return ctx.JSON(http.StatusOK, blueprintResponse)
 }
 
-func (h *Handlers) lintBlueprint(ctx echo.Context, blueprint *BlueprintBody, fixup bool) ([]BlueprintLintItem, error) {
+func (h *Handlers) lintBlueprint(ctx echo.Context, blueprintEntry *db.BlueprintEntry, fixup bool) ([]BlueprintLintItem, error) {
 	lintErrors := []BlueprintLintItem{}
-	if blueprint.Customizations.Openscap != nil {
+	bpBody, err := BlueprintFromEntry(blueprintEntry)
+	if err != nil {
+		return nil, err
+	}
+	if bpBody.Customizations.Openscap != nil {
 		var compl OpenSCAPCompliance
-		var err error
-		if compl, err = blueprint.Customizations.Openscap.AsOpenSCAPCompliance(); err == nil && compl.PolicyId != uuid.Nil {
-			errs, err := h.lintOpenscap(ctx, &blueprint.Customizations, fixup, blueprint.Distribution, compl.PolicyId.String())
+		if compl, err = bpBody.Customizations.Openscap.AsOpenSCAPCompliance(); err == nil && compl.PolicyId != uuid.Nil {
+			var savedPolicy *Customizations
+			if len(blueprintEntry.ServiceSnapshots) != 0 {
+				var snapshots db.ServiceSnapshots
+				if err := json.Unmarshal(blueprintEntry.ServiceSnapshots, &snapshots); err == nil && snapshots.Compliance != nil {
+					if len(snapshots.Compliance.PolicyCustomizations) != 0 {
+						var savedCust Customizations
+						if err := json.Unmarshal(snapshots.Compliance.PolicyCustomizations, &savedCust); err == nil {
+							savedPolicy = &savedCust
+						}
+					}
+				}
+			}
```

Comment:
This makes the code very complex and less readable. In general, having lots of "levels" of code and lots of inner conditions and loops are not a good practice. Someone else might know if we have some kind of "happy path" concept, but if not, you can also try to avoid such big nesting by doing early returns or creating new helper functions.